docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,134 @@
+# ChiselEnum
+
+The ChiselEnum type can be used to reduce the chance of error when encoding mux selectors, opcodes, and functional unit operations. In contrast with`Chisel.util.Enum`, `ChiselEnum` are subclasses of `Data`, which means that they can be used to define fields in `Bundle`s, including in `IO`s.
+
+
+## Functionality and Examples
+
+```scala mdoc
+// Imports used in the following examples
+import chisel3._
+import chisel3.util._
+import chisel3.stage.ChiselStage
+import chisel3.experimental.ChiselEnum
+```
+
+Below we see ChiselEnum being used as mux select for a RISC-V core. While wrapping the object in a package is not required, it is highly recommended as it allows for the type to be used in multiple files more easily. 
+
+```scala mdoc
+// package CPUTypes {
+    object AluMux1Sel extends ChiselEnum {
+        val selectRS1, selectPC = Value
+        /**
+            "selectRS1" -> 0.U,
+            "selectPC"  -> 1.U
+        */
+    }
+// }
+```
+
+Here we see a mux using the AluMux1Sel to select between different inputs. 
+
+```scala mdoc
+import AluMux1Sel._
+
+class AluMux1Bundle extends Bundle {
+        val aluMux1Sel =  Input( AluMux1Sel() )
+        val rs1Out     =  Input(Bits(32.W))
+        val pcOut      =  Input(Bits(32.W))
+        val aluMux1Out = Output(Bits(32.W))
+}
+
+class AluMux1File extends Module {
+    val io = IO(new AluMux1Bundle)
+
+    // Default value for aluMux1Out
+    io.aluMux1Out := 0.U
+
+    switch (io.aluMux1Sel) {
+        is (selectRS1) {
+            io.aluMux1Out  := io.rs1Out
+        }
+        is (selectPC) {
+            io.aluMux1Out  := io.pcOut
+        }
+    }
+}
+```
+```scala mdoc:verilog
+ChiselStage.emitVerilog(new AluMux1File )
+```
+
+ChiselEnum also allows for the user to define variables by passing in the value shown below. Note that the value must be increasing or else 
+
+ > chisel3.internal.ChiselException: Exception thrown when elaborating ChiselGeneratorAnnotation
+
+is thrown during Verilog generation.
+
+```scala 
+object Opcode extends ChiselEnum {
+    val load  = Value(0x03.U) // i load  -> 000_0011
+    val imm   = Value(0x13.U) // i imm   -> 001_0011
+    val auipc = Value(0x17.U) // u auipc -> 001_0111
+    val store = Value(0x23.U) // s store -> 010_0011
+    val reg   = Value(0x33.U) // r reg   -> 011_0011
+    val lui   = Value(0x37.U) // u lui   -> 011_0111
+    val br    = Value(0x63.U) // b br    -> 110_0011
+    val jalr  = Value(0x67.U) // i jalr  -> 110_0111
+    val jal   = Value(0x6F.U) // j jal   -> 110_1111
+}
+```
```

Comment:
Any code block should use the `scala mdoc` tag. This this to get type checked by the compiler and ensure that our documentation isn't heinously busted. 😄